app/views/projects/show.html.erb

```diff
@@ -74,11 +74,11 @@
             <strong><%= t("projects.show.project_access_type") %> </strong>
             <%= @project.project_access_type %>
           </p>
-          <% if !@project.description.nil? %>
+          <% if @project.description.present? && !@project.description.strip.empty? %>
             <p>
               <strong><%= t("projects.show.project_description") %> </strong>
             </p>
-            <div class="projects-details-description"><%= sanitize @project.description %></div>
+            <div class="projects-details-description"><%= sanitize @project.description.gsub("<br>", ' ') %></div>
```

Comment:
Multiple line breaks could be intentional to format the description.

Reply:
Can we just strip start and end of the text?

Reply:
@tanmoysrt I tried that in starting, but (as of my knowledge) the logic remains the same with either we can apply it to all multiple new lines or else we can just go with a non-blank description

Reply:
> @tanmoysrt I tried that in starting, but (as of my knowledge) the logic remains the same with either we can apply it to all multiple new lines or else we can just go with a non-blank description

Update: I looked into this and tried with the `strip` `gsup` method just to remove trailing and leading `<br>` and `spaces` at the end it removes all of the `<br>`.

We might have these changes ...
1. `non-blank` description ( for sure ).
2. only one `new line` or `multiple new lines` for formatting the description.


Please suggest me if there is any other solution.

Reply:
I found one thing: that blank line doesn't means only `<br>` it's actually `<p><br></p>` generated from the editor. So if we want to just trim this part from starting and ending, we can use gsub with regex.

Checkout this URL; this type of regex canbe used to match those parts and replace with empty string after that
https://regex101.com/r/RaZlgR/1

Now we can do that in frontend  in erb or in backend when the user create/update the description of project

Reply:
Yes, thanks for this, able to remove  ending `<p><br></p>`but not starting `<p><br></p>`
I am having trouble with `regex Anchors`

``Approach :`` Used 
```
@project.description.gsub(/\A(<p><br><\/p>)+/, '').gsub(/(<p><br><\/p>)+\z/, '')
```
and 
```
@project.description.gsub(/\A(<p><br><\/p>)+|(<p><br><\/p>)+\z/, ' ')
```

Somehow `/\A(<p><br><\/p>)` is not matching staring `<p><br></p>` ... meanwhile `(<p><br><\/p>)+\z/` able to remove ending `<p><br><\/p>`


`input string`
````
<p><br></p><p><br></p><p><br></p><p>this text is after new lines</p><p><br></p><p>this is again after two new lines which we do not want to trim</p><p><br></p><p><br></p><p><br></p><p><br></p>

````

`output string`
````
<p><br></p><p><br></p><p><br></p><p>this text is after new lines</p><p><br></p><p>this is again after two new lines which we do not want to trim</p>
````

Reply:
@hrshkshri  In ruby it's giving result as expected. https://onecompiler.com/ruby/3zq6wtwgn . That maybe due ti any other issue.